plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineWtpProject.java

```diff
@@ -42,27 +40,32 @@
 import org.eclipse.jst.j2ee.classpathdep.UpdateClasspathAttributeUtil;
 import org.eclipse.ui.actions.WorkspaceModifyOperation;
 import org.eclipse.ui.ide.undo.CreateProjectOperation;
-import org.eclipse.wst.common.project.facet.core.IFacetedProject;
-import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
 
 /**
 * Utility to make a new Eclipse project with the App Engine Standard facets in the workspace.
 */
-class CreateAppEngineStandardWtpProject extends WorkspaceModifyOperation {
+public abstract class CreateAppEngineWtpProject extends WorkspaceModifyOperation {
 
-  private final AppEngineStandardProjectConfig config;
+  private final AppEngineProjectConfig config;
   private final IAdaptable uiInfoAdapter;
   private IFile mostImportant = null;
 
+  public abstract void addAppEngineFacet(IProject newProject, IProgressMonitor monitor) throws CoreException;
+
+  public abstract String getCreateNewProjectOperationLabel();
```

Comment:
Both approaches would work in this case. It's not uncommon for abstract methods to provide labels and descriptions.